client/python/tests/integration/test_no_auth.py

```diff
@@ -1,113 +1,167 @@
-import time
 import uuid
-
+import time
+import os
 import grpc
+
+import pytest
+
+
 from armada_client.client import ArmadaClient
 from armada_client.k8s.io.api.core.v1 import generated_pb2 as core_v1
 from armada_client.k8s.io.apimachinery.pkg.api.resource import (
     generated_pb2 as api_resource,
 )
 
-no_auth_client = ArmadaClient(channel=grpc.insecure_channel(target="127.0.0.1:50051"))
 
+def client() -> ArmadaClient:
+    server_name = os.environ.get("ARMADA_SERVER", "localhost")
+    server_port = os.environ.get("ARMADA_PORT", "50051")
+    server_ssl = os.environ.get("ARMADA_SSL", "false")
 
-def test_submit_job():
-    queue_name = f"queue-{uuid.uuid1()}"
-    job_set_name = f"set-{uuid.uuid1()}"
-    no_auth_client.create_queue(name=queue_name, priority_factor=200)
-    no_auth_client.submit_jobs(
-        queue=queue_name, job_set_id=job_set_name, job_request_items=submit_sleep_job()
-    )
-    # Jobs can must be finished before deleting queue
-    no_auth_client.cancel_jobs(queue=queue_name, job_set_id=job_set_name)
-    no_auth_client.delete_queue(name=queue_name)
-    time.sleep(1)
+    if server_ssl == "true":
+        channel_credentials = grpc.ssl_channel_credentials()
+        return ArmadaClient(
+            channel=grpc.secure_channel(
+                f"{server_name}:{server_port}", channel_credentials
+            )
+        )
 
+    else:
+        return ArmadaClient(
+            channel=grpc.insecure_channel(f"{server_name}:{server_port}")
+        )
 
-def test_submit_job_and_cancel_by_id():
-    queue_name = f"queue-{uuid.uuid1()}"
-    job_set_name = f"set-{uuid.uuid1()}"
-    no_auth_client.create_queue(name=queue_name, priority_factor=200)
-    jobs = no_auth_client.submit_jobs(
-        queue=queue_name, job_set_id=job_set_name, job_request_items=submit_sleep_job()
-    )
-    # Jobs can must be finished before deleting queue
-    no_auth_client.cancel_jobs(job_id=jobs.job_response_items[0].job_id)
-    no_auth_client.delete_queue(name=queue_name)
 
+no_auth_client = client()
 
-def test_submit_job_and_cancel_by_queue_job_id():
-    queue_name = f"queue-{uuid.uuid1()}"
-    job_set_name = f"set-{uuid.uuid1()}"
-    no_auth_client.create_queue(name=queue_name, priority_factor=200)
-    no_auth_client.submit_jobs(
-        queue=queue_name, job_set_id=job_set_name, job_request_items=submit_sleep_job()
+queue_name = f"queue-{uuid.uuid1()}"
```

Comment:
should this (and line 35) be in a pytest setup method and/or fixture?

Reply:
I do not think so, as they are needed across all the functions as globs. I can put them in a better place in the file though.